addons/misra_9.py

```diff
@@ -315,7 +315,7 @@ def parseInitializer(self, root, token):
                                 self.ed.parent.setInitialized(isDesignated)
                             self.ed.parent.initializeChildren()
                     else:
-                        if self.ed.parent != self.root:
+                        if self.root != None and self.ed.parent != self.root:
```

Comment:
I don't know but personally I would not write `!= None`.

When comparing against None I believe `is` is usually used for some reason but it seems overkill here.